selftests/.data/test_statuses.py

```diff
@@ -1,22 +1,308 @@
+
+    def setUp(self):
+        self.log.info('setup pre')
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        self.log.info('test post')
+
+    def tearDown(self):
+        self.log.info('teardown pre')
+        self.fail()
+        self.log.info('teardown post')
+
+
+class WarnSetup(Test):
+
+    def setUp(self):
+        self.log.info('setup pre')
+        self.log.warn('')
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        self.log.info('test post')
+
+    def tearDown(self):
+        self.log.info('teardown pre')
+        self.log.info('teardown post')
+
+
+class Warn(Test):
+
+    def setUp(self):
+        self.log.info('setup pre')
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        self.log.warn('')
+        self.log.info('test post')
+
+    def tearDown(self):
+        self.log.info('teardown pre')
+        self.log.info('teardown post')
+
+
+class WarnTeardown(Test):
+
+    def setUp(self):
+        self.log.info('setup pre')
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        self.log.info('test post')
+
+    def tearDown(self):
+        self.log.info('teardown pre')
+        self.log.warn('')
+        self.log.info('teardown post')
+
+
+class ExitSetup(Test):
+
+    def setUp(self):
+        self.log.info('setup pre')
+        sys.exit(-1)
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        self.log.info('test post')
+
+    def tearDown(self):
+        self.log.info('teardown pre')
+        self.log.info('teardown post')
+
+
+class Exit(Test):
+
+    def setUp(self):
+        self.log.info('setup pre')
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        sys.exit(-1)
+        self.log.info('test post')
+
+    def tearDown(self):
+        self.log.info('teardown pre')
+        self.log.info('teardown post')
+
+
+class ExitTeardown(Test):
+
+    def setUp(self):
+        self.log.info('setup pre')
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        self.log.info('test post')
+
+    def tearDown(self):
+        self.log.info('teardown pre')
+        sys.exit(-1)
+        self.log.info('teardown post')
+
+
+class ExceptionSetup(Test):
+
+    def setUp(self):
+        self.log.info('setup pre')
+        raise ValueError
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        self.log.info('test post')
+
+    def tearDown(self):
+        self.log.info('teardown pre')
+        self.log.info('teardown post')
+
+
+class Exception(Test):
+
+    def setUp(self):
+        self.log.info('setup pre')
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        raise ValueError
```

Comment:
Well this used to be a plain Exception, but ValueError should serve well as well.